modules/e2ee/E2EEncryption.js

```diff
@@ -263,6 +265,25 @@ export class E2EEncryption {
         case 'e2ee.idKey':
             logger.debug(`Participant ${participant.getId()} updated their id key: ${newValue}`);
             break;
+        case 'e2ee.enabled':
+            this._onParticipantE2EEPropertyChanged(participant, newValue);
+            break;
+        }
+    }
+
+    /**
+     * Handles an update in a participant's e2ee.enabled property changed.
+     * If e2ee disabled, frees the olm session and rotates the key.
+     *
+     * @param {JitsiParticipant} participant - The participant.
+     * @param {*} value - The property's previous value.
+     * @private
+     */
+    _onParticipantE2EEPropertyChanged(participant, value) {
+        if (!value && this._enabled) {
+            this._olmAdapter.clearParticipantSession(participant);
```

Comment:
I think we should make this return a boolean, indicating that the rotation is necessary.

Reply:
But is there a situation when the rotation is not necessary?

Reply:
But is there a situation when the rotation is not necessary?

Reply:
It's a bit of a corner case, maybe? I'm thinking a participant who turned e2ee on / off quickly, establishing the session failed. We'd like to avoid rotating too many times I guess. But yeah you can probably skip it.